File: DataUtilities_python/Find_Excel.py
# -*- coding: utf-8 -*-
import os
import pandas as pd

#%% XDS Excel
def Find_Excel(xds):
    
    #%% Find the excel for a given xds file
    
    # Date
    file_name = xds.file_name
    trial_date = file_name.split('_', 1)[0]
    
    # Task
    if xds._lab_data__meta['task_name'] == 'multi_gadget':
        trial_task = 'PG'
    else:
        trial_task = xds._lab_data__meta['task_name']
        
    # Monkey
    monkey_name = xds._lab_data__meta['monkey_name']
    
    # Excel File Name
    excel_name = trial_date + '_' + monkey_name + '_' + trial_task
    
    # File path
    file_path = 'C:/Users/rhpow/Documents/Work/Northwestern/Excel_Data/window_trial_gocue_2/Max_Targets/' 
    # Files & folders
    dir_path = os.listdir(file_path)
    # Table of only files
    files_in_path = list(filter(lambda ii: excel_name in ii, dir_path))
    
    # Selected file
    selec_file = files_in_path[0]
    
    # Load the excel
    try:
        xds_excel = pd.read_excel(file_path + selec_file)
    except:
        print('File cannot be found!')
        xds_excel = 'NaN'
        
    return xds_excel
        



















        
        
        

File: DataUtilities_matlab/Find_Excel.m
function [xds_excel] = Find_Excel(xds)

%% Find the excel for a given xds file

% Date
file_name = xds.meta.rawFileName;
xtra_info = extractAfter(file_name, '_');
Date = erase(file_name, strcat('_', xtra_info));

% Task
if strcmp(xds.meta.task, 'multi_gadget')
    Task = 'PG';
else
    Task = xds.meta.task;
end

% Monkey
Monkey = xds.meta.monkey;

% Excel File Name
excel_name = strcat(Date, '_', Monkey, '_', Task);

% File path
file_path = strcat('C:\Users\rhpow\Documents\Work\Northwestern\Excel_Data\window_trial_gocue\Sorted\');
% Files & folders
dir_path = dir(file_path);
% Table of only files
files_in_path = struct2table(dir_path(~([dir_path.isdir])));

% Selected file
selec_file = contains(files_in_path.name, excel_name);

%% Load the excel file
xds_excel = readtable(strcat(file_path, files_in_path.name{selec_file,1}));

